modules/imaging_browser/php/imaging_session_controlpanel.class.inc

```diff
@@ -96,22 +100,23 @@ class Imaging_Session_ControlPanel
             $params['PCandID'] = $timePoint->getCandID();
             $params['PVL']     = $timePoint->getVisitLabel();
         }
-        $tarchiveIDs = $DB->pselect(
-            "SELECT TarchiveID 
+        // To enable DICOM download: Add pair: tarchiveID, ArchiveLocation
+        $qresult = $DB->pselect(
+            "SELECT TarchiveID, ArchiveLocation 
             FROM tarchive 
             WHERE PatientName LIKE $ID",
             $params
         );
-        $subjectData['tarchiveids'] = $tarchiveIDs;
+
+        $tarIDToTarLoc =array();
+        foreach ($qresult as $k=>$v) {
+            $tarchiveID = $v['TarchiveID'];
+            $tarIDToTarLoc[$tarchiveID] =$v['ArchiveLocation'];
+        }
+        $subjectData['tarchiveIDLoc'] = $tarIDToTarLoc;
 
         $config =& \NDB_Config::singleton();
 
-        //if table is not in database return false to not display in the template
-        $this->tpl_data['mri_param_form_table_exists']
-            = $DB->tableExists('mri_parameter_form');
-        $this->tpl_data['rad_review_table_exists']     = $DB->tableExists(
-            'radiology_review'
-        );
```

Comment:
The tables should exist otherwise the user would not be able to select them in the Configuration module in the first place; so I thought these was no need for these few lines.

Reply:
but you include them as defaults in the patch? What's that going to do if they don't have the tables?

Reply:
it was the case for me for radiology_review table; it just simply won't add it without any error. But maybe it is best I add conditional addition (if table exists) to the patch?